pkg/manager/leaderelection.go

```diff
@@ -35,7 +40,23 @@ func (k *KubeMacPoolManager) waitToStartLeading(poolManger *poolmanager.PoolMana
 	return nil
 }
 
+func (k *KubeMacPoolManager) UpdateLeaderLabel() error {
+	err := k.removeLeaderLabelFromAllKubemacpoolPods()
+	if err != nil {
+		return errors.Wrap(err, "failed cleaning leader label from pods")
+	}
+
+	err = k.AddLeaderLabelToElectedPod()
+	if err != nil {
+		return errors.Wrap(err, "failed adding leader label to elected pod")
+	}
+
+	return nil
+}
```

Comment:
Since we are iterating all the kubemacpool pods we can remove/add label at the same loop, so we can move the code from `removeLeaderLabelAllKubemacpoolPods` here and remove the leader label at all pods except  the one from `POD_NAME`.

Reply:
yes totally